sql/core/src/test/scala/org/apache/spark/sql/StringLiteralCoalescingSuite.scala

```diff
@@ -0,0 +1,786 @@
+  // LIKE and Pattern Matching Tests
+  // ========================================================================
+
+  test("string coalescing in LIKE patterns") {
+    val testCases = Seq(
+      // Coalescing with underscore wildcard - match
+      ("SELECT 'test_value' LIKE 'test' '_value'", true),
+      // Coalescing with underscore wildcard - no match
+      ("SELECT 'test_value' LIKE 'test' '_xyz'", false),
+      // Coalescing with percent wildcard
+      ("SELECT 'prefix_middle_suffix' LIKE 'prefix' '%' 'suffix'", true),
+      // ESCAPE clause with coalescing
+      ("SELECT 'test%value' LIKE 'test' '\\%' 'value' ESCAPE '\\\\'", true)
+    )
+
+    testCases.foreach { case (query, expected) =>
+      checkAnswer(sql(query), Row(expected))
+    }
+  }
+
+  test("string coalescing in table options") {
+    withTable("t") {
+      withTempPath { dir =>
+        val path1 = s"${dir.getAbsolutePath}/part"
+        val path2 = "1"
+        // Test that LOCATION paths can be coalesced
+        sql(s"CREATE TABLE t (a STRING, b STRING) USING parquet LOCATION '$path1' '$path2'")
+        assert(spark.catalog.tableExists("t"))
+        val location = spark.sessionState.catalog.getTableMetadata(
+          spark.sessionState.sqlParser.parseTableIdentifier("t")).location
+        assert(location.toString.contains("part1"))
+      }
+    }
+  }
+
+  test("string coalescing in SHOW TABLES LIKE pattern") {
+    val tableNames = Seq("test_table_123", "test_table_456", "other_table")
+    withTable(tableNames: _*) {
+      tableNames.foreach { tableName =>
+        sql(s"CREATE TABLE $tableName (id INT)")
+      }
+
+      // The pattern is coalesced into 'test_table_*' (regex pattern where * matches any chars)
+      // SHOW TABLES returns: namespace, tableName, isTemporary
+      checkAnswer(
+        sql("SHOW TABLES LIKE 'test' '_table_' '*'"),
+        Seq(
+          Row("default", "test_table_123", false),
+          Row("default", "test_table_456", false)
+        )
+      )
+    }
+  }
+
+  test("string coalescing across multiple lines") {
+    val result = sql("""
+      SELECT 'line'
+             'one'
+             'two'
+    """).collect().head.getString(0)
+    assert(result == "lineonetwo")
+  }
+
+  test("string coalescing in WHERE clause") {
+    withTable("t") {
+      sql("CREATE TABLE t (name STRING) USING parquet")
+      sql("INSERT INTO t VALUES ('helloworld'), ('hello'), ('world')")
+
+      checkAnswer(
+        sql("SELECT * FROM t WHERE name = 'hello' 'world'"),
+        Row("helloworld")
+      )
+    }
+  }
+
+  // ========================================================================
+  // R-String (Raw String) Coalescing Tests
+  // ========================================================================
+
+  test("R-string detection - basic cases and edge cases") {
+    val testCases = Seq(
+      // Basic cases: uppercase and lowercase R with single/double quotes
+      ("""SELECT R'\n'""", """\n"""),
```

Comment:
The test is incorrect. `"""\n"""` in scala still does escape, what we really want is `raw"\n"`, as Scala has a similar raw string syntax as SQL.